packages/visual-layer/src/helpers/index.js

```diff
@@ -471,18 +471,26 @@ export const renderLayer = (context) => {
     }
 };
 
+const transformResolverPredicates = [
+    encodingFieldInf => !encodingFieldInf.xField || !encodingFieldInf.yField,
+    (encodingFieldInf, context, groupByField) => {
+        const fieldsConfig = context.data().getFieldsConfig();
+        return !groupByField || getObjProp(fieldsConfig[groupByField], 'def', 'type') === FieldType.MEASURE;
+    },
+    encodingFieldInf => encodingFieldInf.xFieldType === FieldType.DIMENSION &&
+        encodingFieldInf.yFieldType === FieldType.DIMENSION,
+    (encodingFieldInf, context, groupByField) => {
+        const dimensionField = ['xField', 'yField'].find(type =>
+                encodingFieldInf[`${type}Type`] === FieldType.DIMENSION);
+        return (dimensionField && encodingFieldInf[dimensionField] === groupByField);
+    }
+];
```

Comment:
Use this array as a function so that you can keep the variables in a closure and call the conditions on the return value of the array